### ISSUE TITLE
TypeVar with constraints not assignable to Union that have same types

### ISSUE THREAD

**Issue by cleoold** [microsoft/pyright#1245]
**Describe the bug**
Title.

**To Reproduce**
Type the following code in vscode editor.

**Expected behavior**
No error should happen.

**Screenshots or Code**
```python
def func(a: Union[int, float]): ...

_N = TypeVar('_N', int, float)

def bar(a: _N):
    return func(a)
           ~~~~~^
```

This is the error produced (which did not happen before the update):
```py
Argument of type "_N" cannot be assigned to parameter "a" of type "int | float" in function "func"
  Type "_N" cannot be assigned to type "int | float"
    Type "_N" cannot be assigned to type "int"
    Type "_N" cannot be assigned to type "float"Pyright (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
vscode extension 1.1.92

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. I've fixed the issue and added test cases to cover it. The fix will be in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.93, which I just published. It will also be included in the next release of Pylance.
